src/summary/summary/celery_worker.py

```diff
@@ -0,0 +1,120 @@
+"""Celery workers."""
+
+import json
+import tempfile
+from pathlib import Path
+
+import openai
+from celery import Celery
+from celery.utils.log import get_task_logger
+from minio import Minio
+from requests import Session
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+
+from .config import Settings
+from .prompt import get_instructions
+
+settings = Settings()
+
+
+logger = get_task_logger(__name__)
+
+celery = Celery(
+    __name__,
+    broker=settings.celery_broker_url,
+    backend=settings.celery_result_backend,
+    broker_connection_retry_on_startup=True,
+)
+
+
+def save_audio_stream(audio_stream, chunk_size=32 * 1024):
+    """Save an audio stream to a temporary OGG file."""
+    with tempfile.NamedTemporaryFile(suffix=".ogg", delete=False) as tmp:
+        tmp.writelines(audio_stream.stream(chunk_size))
+        return Path(tmp.name)
+
+
+def create_retry_session():
```

Comment:
Nit: it's not only retry, there is authentication too

Reply:
yes, I refactored the code